it/uk/gov/hmrc/agentmappingfrontend/controllers/MappingControllerISpec.scala

```diff
@@ -63,263 +71,78 @@ class MappingControllerISpec extends BaseControllerISpec with AuthStubs {
-      redirectLocation(result) shouldBe Some(routes.MappingController.showEnterUtr.url)
-      result.session(request).get("mappingArn") shouldBe Some("TARN0000001")
-    }
-
-    "redirect and add arn to session when arn in hyphen pattern is entered" in {
-      givenUserIsAuthenticated(eligibleAgent)
-      val request = createRequest("TARN-000-0001")
-      val result = callEndpointWith(request)
-      redirectLocation(result) shouldBe Some(routes.MappingController.showEnterUtr.url)
-      result.session(request).get("mappingArn") shouldBe Some("TARN0000001")
-    }
-
-    "re-enter arn since invalid" in {
-      givenUserIsAuthenticated(eligibleAgent)
-      val result = callEndpointWith(createRequest("invalidArn"))
-      status(result) shouldBe 200
-      checkHtmlResultContainsMsgs(result,"error.arn.invalid")
-    }
-
-    "re-enter arn if an arn entered with invalid format" in {
-      givenUserIsAuthenticated(eligibleAgent)
-      val result = callEndpointWith(createRequest("TARN-0000-001"))
-      status(result) shouldBe 200
-      checkHtmlResultContainsMsgs(result,"error.arn.invalid")
-    }
-
-    "re-enter arn since empty" in {
-      givenUserIsAuthenticated(eligibleAgent)
-      val result = callEndpointWith(createRequest(""))
-      status(result) shouldBe 200
-      checkHtmlResultContainsMsgs(result,"error.arn.blank")
-    }
-  }
-
-  "show enter-utr" should {
-    def request = fakeRequest(GET, routes.MappingController.showEnterUtr.url)
-
-    "enter-utr arn found in session" in {
-      givenUserIsAuthenticated(eligibleAgent)
-
-      val result = callEndpointWith(request.withSession(("mappingArn", "TARN0000001")))
-      status(result) shouldBe 200
-    }
-
-    "redirect noArn go back to enter-account-number" in {
-      givenUserIsAuthenticated(eligibleAgent)
 
-      val result = callEndpointWith(request)
-      redirectLocation(result) shouldBe Some(routes.MappingController.showEnterAccountNo.url)
-    }
-  }
-
-  "submit add code" should {
-
-    val endpoint = "/agent-mapping/enter-utr"
-
-    behave like anEndpointReachableIfSignedInWithEligibleEnrolment(
-      POST,
-      endpoint,
-      expectCheckAgentRefCodeAudit = false)(callEndpointWith)
-
-    "redirect to complete if the user enters an ARN and UTR that match the known facts for SA" in {
-      givenUserIsAuthenticated(eligibleAgent)
-      mappingIsCreated(Utr("2000000000"), Arn("TARN0000001"))
-      val request = fakeRequest(POST, endpoint).withFormUrlEncodedBody("utr.value" -> "2000000000")
-        .withSession(("mappingArn", "TARN0000001"))
-      val result = callEndpointWith(request)
+        status(result) shouldBe 303
+        result.session.get("mappingArn").get shouldBe "TARN0000001"
+        redirectLocation(result).get should include(routes.MappingController.complete(id = "").url)
 
-      status(result) shouldBe 303
-      redirectLocation(result).get shouldBe routes.MappingController.complete().url
+        givenAuthorisedFor(serviceName)
```

Comment:
Lines 99 to 103 here look like an indirect way of testing two things at the same time: 
a) that the ID has been removed from the repository. Could we just assert directly that it's not in the repository at the end of this test?
and b) testing that calling the `/start-submit` with an unknown ID will redirect to the start page. Could we have this as a separate test (if not already)?